riscv/src/reachability.rs

```diff
@@ -49,8 +48,7 @@ pub fn filter_reachable_from(
             panic!();
         };
         for referenced in &new_references {
-            if !queued_labels.contains(referenced) && !processed_labels.contains(referenced) {
-                label_queue.push(referenced);
+            if !processed_labels.contains(referenced) {
```

Comment:
If you don't check that is already in `queued_labels` you might insert it twice.

Reply:
Before the first instance of the same item is visited I mean.

Reply:
It is a Set, there can be only one.